packages/parser/src/grammar.ts

```
@@ -35,6 +35,101 @@ import {
   whitespace,
 } from "./whitespace-utils";
 
+
+/**
+ * Bounded expression parser for grammar rules
+ * This parser stops at newlines or the next rule definition to prevent consuming the next rule
+ */
+const grammarRuleExpression: Parser<any> = (input: string, pos) => {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)